toolboxes/mri_core/mri_core_girf_correction.cpp

```diff
@@ -1,160 +1,282 @@
+                                                      float girf_sampling_time, float TE) // Times are in us
+        {
 
-            size_t nlines = gradients.get_size(0);
-//            size_t nlines_new = size_t(nlines * girf_sampling_time / gradient_sampling_time);
-            size_t nlines_new = girf_kernel.get_size(0);
+            using namespace std;
 
+            size_t nlines = gradients.get_size(0);
+            size_t nlines_new = girf_kernel.get_size(0) * std::round(girf_sampling_time / gradient_sampling_time); //std::max(girf_kernel.get_size(0) * girf_sampling_time / gradient_sampling_time,nlines * std::round(girf_sampling_time / gradient_sampling_time)); // This logic here may need to be checked
             size_t nbatches = gradients.get_size(1);
 
-            hoNDArray<vector_td<float,D>> result(gradients.get_dimensions());
+            //auto padded_kernel=GIRF::zeropadding(girf_kernel,int(girf_sampling_time/gradient_sampling_time));
 
-            for (int batch = 0; batch < nbatches; batch++) {
+            auto padded_kernel = pad<std::complex<float>, 1>(uint64d1(nlines_new), girf_kernel, 0);
+            //auto padded_kernel = GIRF::zeropadding(girf_kernel, 5);
+            hoNDArray<vector_td<float, D>> result(gradients.get_dimensions());
 
-                hoNDArray<std::complex<float>> rotated(nlines_new, 3);
-                std::fill(rotated.begin(), rotated.end(), 0);
+            hoNDArray<float> filter;
+            using namespace Gadgetron::Indexing;
+
+            // Generate Hanning window filter (using ACW parameters) and normalize the max to 1
+            Gadgetron::generate_symmetric_filter(400, filter, ISMRMRD_FILTER_HANNING, 1.5, 400);
+            auto fmax = max(filter);
+            filter /= fmax;
 
+            hoNDArray<std::complex<float>> rotated(nlines_new, 3);
 
-                for (size_t i = 0; i < nlines; i++) {
+            for (int batch = 0; batch < nbatches; batch++)
+            {
+                std::fill(rotated.begin(), rotated.end(), 0);
+                for (size_t i = 0; i < nlines; i++)
+                {
                     arma::fvec3 gradient_vector(arma::fill::zeros);
                     for (size_t k = 0; k < D; k++)
-                        gradient_vector[k] = gradients(i,batch)[k];
+                        gradient_vector[k] = gradients(i, batch)[k];
 
                     gradient_vector = rotation_matrix * gradient_vector;
                     for (size_t k = 0; k < 3; k++)
                         rotated(i, k) = gradient_vector[k];
                 }
+                float maxTx;
+                float minTx;
+                auto temp = real(permute(rotated, {0, 1}));
+                Gadgetron::maxValue(hoNDArray<float>(temp(slice, 0)), maxTx);
+                Gadgetron::minValue(hoNDArray<float>(temp(slice, 0)), minTx);
+
+                if (maxTx > 100 || minTx < -100)
+                    GERROR("What the heck");
```

Comment:
Not useful. Feels like a debug comment. If it is actually a failstate, throw an exception.

Reply:
Removed.